trace/trace.go

```diff
@@ -145,13 +154,25 @@ func WithSampler(sampler Sampler) StartOption {
 	}
 }
 
+// WithBufferLimit makes new spans with a custom buffer limit.
+// This limits variable trace span data: Message Events, Links, and Annotations.
+// The default is 1000 (trace.DefaultBufferLimit) and the minimum is 1.
+func WithBufferLimit(limit int) StartOption {
+	return func(o *StartOptions) {
+		o.BufferLimit = limit
+		if o.BufferLimit <= 0 {
+			o.BufferLimit = 1
+		}
+	}
+}
+
 // StartSpan starts a new child span of the current span in the context. If
 // there is no span in the context, creates a new trace and span.
 //
 // Returned context contains the newly created span. You can use it to
 // propagate the returned span in process.
 func StartSpan(ctx context.Context, name string, o ...StartOption) (context.Context, *Span) {
-	var opts StartOptions
+	opts := StartOptions{BufferLimit: DefaultBufferLimit}
```

Comment:
Don't set the default here.

In startSpanInternal, set the default if opts.BufferLimit is zero.